GitUI/CommandsDialogs/BrowseDialog/DashboardControl/UserRepositoriesList.cs

```diff
@@ -215,31 +229,72 @@ public void ShowRecentRepositories()
             {
                 listView1.BeginUpdate();
                 listView1.Items.Clear();
-                if (recentRepositories.Count == 0 && favouriteRepositories.Count == 0)
+                if (recentRepositories.Count > 0 || favouriteRepositories.Count > 0)
                 {
-                    return;
+                    listView1.TileSize = GetTileSize(recentRepositories, favouriteRepositories);
+                    Debug.WriteLine($"Tile size: {listView1.TileSize}");
                 }
 
-                listView1.TileSize = GetTileSize(recentRepositories, favouriteRepositories);
-                Debug.WriteLine($"Tile size: {listView1.TileSize}");
-
                 _hasInvalidRepos = false;
-                BindRepositories(recentRepositories);
-                BindRepositories(favouriteRepositories);
+
+                SetupGroups(recentRepositories.Concat(favouriteRepositories));
+                BindRepositories(recentRepositories, isFavourite: false);
+                BindRepositories(favouriteRepositories, isFavourite: true);
             }
             finally
             {
-                listView1.SetGroupState(ListViewGroupState.Collapsible);
                 listView1.EndUpdate();
             }
 
-            void BindRepositories(IReadOnlyList<RecentRepoInfo> repos)
+            void SetupGroups(IEnumerable<RecentRepoInfo> repositories)
```

Comment:
Can we turn this into an extension method, so we can test it?
Or could/should this be moved into `ExListView` because it is an implementation detail?

Reply:
moved to ExListView